astyanax-cassandra/src/main/java/com/netflix/astyanax/serializers/ComparatorType.java

```diff
@@ -15,45 +15,43 @@
  */
 package com.netflix.astyanax.serializers;
 
-import com.netflix.astyanax.*;
+import com.netflix.astyanax.Serializer;
+import com.netflix.astyanax.shaded.org.apache.cassandra.db.marshal.ShadedTypeParser;
+
+import static com.netflix.astyanax.shaded.org.apache.cassandra.db.marshal.ShadedTypeParser.SHADED_PREFIX;
 
 /**
  * @author: peter
  */
 public enum ComparatorType {
 
```

Comment:
Would this comparator come from C*? or client side?

Reply:
Doesn't really matter. This comparator is just responsible for returning the correct serializer regardless of whether a given type's class is in the original Cassandra package (e.g. handling messages to/from C* cluster) or the shaded Cassandra package (e.g. code within Astyanax). It's the core guts of what makes Astayanax's shaded cassandra-all talk to other processes which use unshaded original cassandra-all.